staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -83,12 +84,19 @@ type watchCacheEvent struct {
 // in different List/Watch requests), in the underlying store we are
 // keeping structs (key, object, labels, fields).
 type storeElement struct {
+	RV     uint64
```

Comment:
Why do you want to add RV?

At any given point in time, there can't be more objects with the same Key, so we can just use Key to compare objects.

Reply:
@wojtek-t thanks for the initial review. Maybe I've misunderstood something, but this is my understanding so far on a high level:
- The btree backed store that we create is ordered on the resource version so that we can iterate over the objects we have in-order to serve a LIST request.
- We call `ListSince(rvWeWantFrom)` which returns the elements since a particular rv (we can also add in logic to only return upto `limit` items)

Please let me know in case I've misunderstood your intent here. Thanks again!

(the above is likely to change once we have indexes in place and we make a clone to serve continuation requests, but in principle, this is my understanding)

Reply:
> The btree backed store that we create is ordered on the resource version so that we can iterate over the objects we have in-order to serve a LIST request.

LIST requests are not sorted by RV now, so we don't need it.

> We call ListSince(rvWeWantFrom) which returns the elements since a particular rv (we can also add in logic to only return upto limit items)

It's not "Since", we want to be able to list items that reflect the state that we had in rv=X.

Reply:
Ohh, okay I think I understand it now. Thank you, I'll get started working on the changes.